tests/compiler/class-overloading-cast.ts

```diff
@@ -0,0 +1,16 @@
+class A<T> {
+  foo(a: T): void {
+    trace(`class A ${a}`);
+  }
```

Comment:
How about return string?
```ts
foo(a: T): string {
    return "A";
}
```
and same for `B#foo` but return `"B"`.
And later call this functions with and compare with expected values with assert